In [2]:
%cd ..
import os
from engine.utils.metrics.metric import calculate_cod_metrics

os.chdir("/Users/felix/Desktop/UCOD-DPL")

result = calculate_cod_metrics(
    "datasets/RefCOD/TE-COD10K/gt",
    "results/preds/TE-COD10K",
)

{k: round(float(v), 4) for k, v in result.items()}

/Users/felix/Desktop/UCOD-DPL


100%|██████████| 2026/2026 [06:28<00:00,  5.21it/s]


{'E_MAX': 0.9185,
 'E_MEAN': 0.9159,
 'F_MAX': 0.7813,
 'F_MEAN': 0.7787,
 'SMeasure': 0.8336,
 'MAE': 0.0305,
 'WFM': 0.763}


| Dataset   | ACC    | mIOU   | E_MAX  | E_MEAN | F_MAX  | F_MEAN | SMeasure | MAE    | WFM    |
| --------- | ------ | ------ | ------ | ------ | ------ | ------ | -------- | ------ | ------ |
| CHAMELEON | 0.9688 | 0.7538 | 0.9332 | 0.9306 | 0.8409 | 0.8383 | 0.8644   | 0.0312 | 0.8255 |
| TE-CAMO   | 0.9240 | 0.6547 | 0.8659 | 0.8635 | 0.7824 | 0.7802 | 0.7949   | 0.0760 | 0.7489 |
| TE-COD10K | 0.9695 | 0.6798 | 0.9185 | 0.9159 | 0.7813 | 0.7787 | 0.8336   | 0.0305 | 0.7630 |
| NC4K      | 0.9575 | 0.7343 | 0.9255 | 0.9229 | 0.8378 | 0.8352 | 0.8496   | 0.0425 | 0.8178 |
